test/integration/analytical_policy.hpp

```diff
@@ -78,8 +78,9 @@ using yields = std::pair<micm::Species, double>;
 
 using SparseMatrixTest = micm::SparseMatrix<double>;
 
-template<class BuilderPolicy>
-void test_analytical_troe(BuilderPolicy& builder, double tolerance = 1e-8)
+
+template<class BuilderPolicy, class PrepareFunc, class PostpareFunc>
+void test_analytical_troe(BuilderPolicy& builder, PrepareFunc prepare_for_solve, PostpareFunc postpare_for_solve, double tolerance = 1e-8)
```

Comment:
Do we need templates for the before and after functions? Couldn't we just have arguements with default values so that the CPU tests don't have to pass in empty functions?